naga/src/proc/constant_evaluator.rs

```diff
@@ -1186,15 +1196,20 @@ impl<'a> ConstantEvaluator<'a> {
                 )
             }
             crate::MathFunction::Step => {
-                component_wise_float!(self, span, [arg, arg1.unwrap()], |edge, x| {
-                    Ok([if edge <= x { 1.0 } else { 0.0 }])
-                })
+                //component_wise_float!(self, span, [arg, arg1.unwrap()], |edge, x| {
+                //    Ok([if edge <= x { 1.0 } else { 0.0 }])
+                //})
+                todo!()
```

Comment:
TODO

Reply:
We can match like in the other places? Looking at the spec we should error if the types of the 2 args are different.

Reply:
Sorry not sure I follow, the macro is a little hairy.

Reply:
Should be improved now, still not 100% certain on its correctness.